File: Java/uuid_utils/uuid_utils_test.java
package uuid_utils;

public class uuid_utils_test {
    private static int testsPassed = 0;
    private static int testsFailed = 0;

    public static void main(String[] args) {
        System.out.println("=== UUID Utilities Test Suite ===\n");
        testV4Generation();
        testV4Compact();
        testV4Upper();
        testV3Generation();
        testV5Generation();
        testRandomString();
        testRandomAlphanumeric();
        testRandomNumeric();
        testRandomHex();
        testRandomPassword();
        testValidation();
        testV4Validation();
        testFormatConversion();
        testParsing();
        testVersionAndVariant();
        testShortId();
        testNanoId();
        testUlid();
        System.out.println("\n=== Test Summary ===");
        System.out.println("Passed: " + testsPassed);
        System.out.println("Failed: " + testsFailed);
        System.out.println("Total:  " + (testsPassed + testsFailed));
        if (testsFailed > 0) System.exit(1);
    }

    private static void testV4Generation() {
        System.out.println("Testing UUID v4 generation...");
        String uuid = mod.generateV4();
        assertTrue(uuid != null, "UUID should not be null");
        assertTrue(uuid.length() == 36, "UUID should be 36 characters");
        assertTrue(uuid.matches("^[0-9a-f]{8}-[0-9a-f]{4}-4[0-9a-f]{3}-[89ab][0-9a-f]{3}-[0-9a-f]{12}$"),
                "UUID should match v4 format");
        String uuid2 = mod.generateV4();
        assertTrue(!uuid.equals(uuid2), "Two UUIDs should be different");
        System.out.println("  ✓ UUID v4 generation tests passed");
    }

    private static void testV4Compact() {
        System.out.println("Testing UUID v4 compact format...");
        String uuid = mod.generateV4Compact();
        assertTrue(uuid != null, "Compact UUID should not be null");
        assertTrue(uuid.length() == 32, "Compact UUID should be 32 characters");
        assertTrue(!uuid.contains("-"), "Compact UUID should not contain dashes");
        assertTrue(uuid.matches("^[0-9a-f]{32}$"), "Compact UUID should be lowercase hex");
        System.out.println("  ✓ UUID v4 compact tests passed");
    }

    private static void testV4Upper() {
        System.out.println("Testing UUID v4 uppercase format...");
        String uuid = mod.generateV4Upper();
        assertTrue(uuid != null, "Upper UUID should not be null");
        assertTrue(uuid.length() == 36, "Upper UUID should be 36 characters");
        assertTrue(uuid.equals(uuid.toUpperCase()), "Upper UUID should be uppercase");
        System.out.println("  ✓ UUID v4 uppercase tests passed");
    }

    private static void testV3Generation() {
        System.out.println("Testing UUID v3 generation...");
        String uuid1 = mod.generateV3(mod.NAMESPACE_DNS, "example.com");
        String uuid2 = mod.generateV3(mod.NAMESPACE_DNS, "example.com");
        assertTrue(uuid1 != null, "v3 UUID should not be null");
        assertTrue(uuid1.equals(uuid2), "Same namespace+name should produce same v3 UUID");
        assertTrue(mod.getVersion(uuid1) == 3, "Should be version 3");
        String uuid3 = mod.generateV3(mod.NAMESPACE_DNS, "example.org");
        assertTrue(!uuid1.equals(uuid3), "Different names should produce different UUIDs");
        System.out.println("  ✓ UUID v3 generation tests passed");
    }

    private static void testV5Generation() {
        System.out.println("Testing UUID v5 generation...");
        String uuid1 = mod.generateV5(mod.NAMESPACE_URL, "https://example.com");
        String uuid2 = mod.generateV5(mod.NAMESPACE_URL, "https://example.com");
        assertTrue(uuid1 != null, "v5 UUID should not be null");
        assertTrue(uuid1.equals(uuid2), "Same namespace+name should produce same v5 UUID");
        assertTrue(mod.getVersion(uuid1) == 5, "Should be version 5");
        System.out.println("  ✓ UUID v5 generation tests passed");
    }

    private static void testRandomString() {
        System.out.println("Testing random string generation...");
        String str = mod.randomString(16, null);
        assertTrue(str != null, "Random string should not be null");
        assertTrue(str.length() == 16, "Random string should have correct length");
        String hex = mod.randomString(32, "0123456789ABCDEF");
        assertTrue(hex.matches("^[0-9A-F]+$"), "Should only contain hex characters");
        String str2 = mod.randomString(16, null);
        assertTrue(!str.equals(str2), "Two random strings should likely be different");
        try {
            mod.randomString(-1, null);
            assertTrue(false, "Should throw exception for negative length");
        } catch (IllegalArgumentException e) {
            assertTrue(true, "Should throw exception for negative length");
        }
        System.out.println("  ✓ Random string tests passed");
    }

    private static void testRandomAlphanumeric() {
        System.out.println("Testing random alphanumeric generation...");
        String str = mod.randomAlphanumeric(20);
        assertTrue(str != null, "Alphanumeric should not be null");
        assertTrue(str.length() == 20, "Should have correct length");
        assertTrue(str.matches("^[A-Za-z0-9]+$"), "Should be alphanumeric");
        System.out.println("  ✓ Random alphanumeric tests passed");
    }

    private static void testRandomNumeric() {
        System.out.println("Testing random numeric generation...");
        String str = mod.randomNumeric(10);
        assertTrue(str != null, "Numeric should not be null");
        assertTrue(str.length() == 10, "Should have correct length");
        assertTrue(str.matches("^[0-9]+$"), "Should be numeric only");
        System.out.println("  ✓ Random numeric tests passed");
    }

    private static void testRandomHex() {
        System.out.println("Testing random hex generation...");
        String str = mod.randomHex(32);
        assertTrue(str != null, "Hex should not be null");
        assertTrue(str.length() == 32, "Should have correct length");
        assertTrue(str.matches("^[0-9a-f]+$"), "Should be lowercase hex");
        System.out.println("  ✓ Random hex tests passed");
    }

    private static void testRandomPassword() {
        System.out.println("Testing random password generation...");
        String pwd = mod.randomPassword(16);
        assertTrue(pwd != null, "Password should not be null");
        assertTrue(pwd.length() == 16, "Should have correct length");
        boolean hasUpper = pwd.matches(".*[A-Z].*");
        boolean hasLower = pwd.matches(".*[a-z].*");
        boolean hasDigit = pwd.matches(".*[0-9].*");
        boolean hasSpecial = pwd.matches(".*[!@#$%^&*].*");
        assertTrue(hasUpper, "Password should have uppercase");
        assertTrue(hasLower, "Password should have lowercase");
        assertTrue(hasDigit, "Password should have digit");
        assertTrue(hasSpecial, "Password should have special char");
        System.out.println("  ✓ Random password tests passed");
    }

    private static void testValidation() {
        System.out.println("Testing UUID validation...");
        assertTrue(mod.isValidUUID("550e8400-e29b-41d4-a716-446655440000"), "Valid UUID should be valid");
        assertTrue(mod.isValidUUID("550e8400e29b41d4a716446655440000"), "Compact UUID should be valid");
        assertTrue(mod.isValidUUID("550E8400-E29B-41D4-A716-446655440000"), "Uppercase UUID should be valid");
        assertTrue(!mod.isValidUUID("invalid"), "Invalid string should not be valid");
        assertTrue(!mod.isValidUUID(""), "Empty string should not be valid");
        assertTrue(!mod.isValidUUID(null), "Null should not be valid");
        assertTrue(!mod.isValidUUID("550e8400-e29b-41d4-a716"), "Short UUID should not be valid");
        System.out.println("  ✓ UUID validation tests passed");
    }

    private static void testV4Validation() {
        System.out.println("Testing UUID v4 validation...");
        String v4 = mod.generateV4();
        assertTrue(mod.isValidV4(v4), "Generated v4 should be valid v4");
        String v5 = mod.generateV5(mod.NAMESPACE_DNS, "test");
        assertTrue(!mod.isValidV4(v5), "v5 UUID should not be valid v4");
        assertTrue(!mod.isValidV4("invalid"), "Invalid string should not be valid v4");
        System.out.println("  ✓ UUID v4 validation tests passed");
    }

    private static void testFormatConversion() {
        System.out.println("Testing format conversion...");
        String standard = "550e8400-e29b-41d4-a716-446655440000";
        String compact = "550e8400e29b41d4a716446655440000";
        assertTrue(mod.toCompact(standard).equals(compact), "toCompact should remove dashes");
        assertTrue(mod.toStandard(compact).equals(standard), "toStandard should add dashes");
        assertTrue(mod.toUpperCase(standard).equals("550E8400-E29B-41D4-A716-446655440000"), "toUpperCase should uppercase");
        assertTrue(mod.toCompact("invalid") == null, "Invalid UUID should return null");
        System.out.println("  ✓ Format conversion tests passed");
    }

    private static void testParsing() {
        System.out.println("Testing UUID parsing...");
        String uuidStr = mod.generateV4();
        java.util.UUID uuid = mod.parse(uuidStr);
        assertTrue(uuid != null, "Parsed UUID should not be null");
        assertTrue(uuid.toString().equals(uuidStr), "Parsed UUID should match original");
        assertTrue(mod.parse("invalid") == null, "Invalid UUID should return null");
        System.out.println("  ✓ UUID parsing tests passed");
    }

    private static void testVersionAndVariant() {
        System.out.println("Testing version and variant...");
        String v4 = mod.generateV4();
        assertTrue(mod.getVersion(v4) == 4, "v4 should have version 4");
        String v3 = mod.generateV3(mod.NAMESPACE_DNS, "test");
        assertTrue(mod.getVersion(v3) == 3, "v3 should have version 3");
        String v5 = mod.generateV5(mod.NAMESPACE_DNS, "test");
        assertTrue(mod.getVersion(v5) == 5, "v5 should have version 5");
        assertTrue(mod.getVariant(v4) == 1, "RFC4122 variant should be 1");
        assertTrue(mod.getVersion("invalid") == -1, "Invalid UUID should return -1");
        System.out.println("  ✓ Version and variant tests passed");
    }

    private static void testShortId() {
        System.out.println("Testing short ID generation...");
        String id = mod.shortId(8);
        assertTrue(id != null, "Short ID should not be null");
        assertTrue(id.length() > 0, "Short ID should not be empty");
        String id2 = mod.shortId(8);
        assertTrue(!id.equals(id2), "Two short IDs should be different");
        System.out.println("  ✓ Short ID tests passed");
    }

    private static void testNanoId() {
        System.out.println("Testing Nano ID generation...");
        String nano = mod.nanoId();
        assertTrue(nano != null, "Nano ID should not be null");
        assertTrue(nano.length() == 21, "Default Nano ID should be 21 characters");
        String nano10 = mod.nanoId(10);
        assertTrue(nano10.length() == 10, "Custom Nano ID should have correct length");
        System.out.println("  ✓ Nano ID tests passed");
    }

    private static void testUlid() {
        System.out.println("Testing ULID generation...");
        String ulid1 = mod.ulid();
        String ulid2 = mod.ulid();
        assertTrue(ulid1 != null, "ULID should not be null");
        assertTrue(ulid1.length() == 26, "ULID should be 26 characters");
        assertTrue(!ulid1.equals(ulid2), "Two ULIDs should be different");
        System.out.println("  ✓ ULID tests passed");
    }

    private static void assertTrue(boolean condition, String message) {
        if (condition) {
            testsPassed++;
        } else {
            testsFailed++;
            System.out.println("    ✗ FAILED: " + message);
        }
    }
}

File: TypeScript/uuid_utils/uuid_utils_test.ts
/**
 * UUID Utilities Test Suite
 * 
 * Comprehensive tests for the UUID utilities module.
 * Run with: npx ts-node uuid_utils_test.ts
 * Or compile: tsc uuid_utils_test.ts && node uuid_utils_test.js
 */

import {
  v4,
  v1,
  compact,
  uppercase,
  isValid,
  parse,
  getVersion,
  getVariant,
  toCompact,
  toStandard,
  equals,
  nanoId,
  shortId,
  randomString,
  randomPassword,
  UuidUtils,
  CHARSETS,
} from './mod';

// Test result tracking
let passed = 0;
let failed = 0;

function test(name: string, fn: () => void): void {
  try {
    fn();
    console.log(`✓ ${name}`);
    passed++;
  } catch (error) {
    console.error(`✗ ${name}`);
    console.error(`  Error: ${error}`);
    failed++;
  }
}

function assertEqual(actual: any, expected: any, message?: string): void {
  if (actual !== expected) {
    throw new Error(`${message || 'Assertion failed'}: expected ${expected}, got ${actual}`);
  }
}

function assertTrue(value: boolean, message?: string): void {
  if (!value) {
    throw new Error(message || 'Expected true, got false');
  }
}

function assertFalse(value: boolean, message?: string): void {
  if (value) {
    throw new Error(message || 'Expected false, got true');
  }
}

function assertMatch(value: string, pattern: RegExp, message?: string): void {
  if (!pattern.test(value)) {
    throw new Error(`${message || 'Pattern match failed'}: ${value} does not match ${pattern}`);
  }
}

// ==================== UUID v4 Tests ====================

test('v4() generates valid UUID', () => {
  const uuid = v4();
  assertTrue(isValid(uuid), 'Generated UUID should be valid');
  assertEqual(getVersion(uuid), 4, 'Should be version 4');
});

test('v4() generates unique UUIDs', () => {
  const uuids = new Set<string>();
  for (let i = 0; i < 100; i++) {
    uuids.add(v4());
  }
  assertEqual(uuids.size, 100, 'All 100 UUIDs should be unique');
});

test('v4() generates correct format', () => {
  const uuid = v4();
  assertMatch(uuid, /^[0-9a-f]{8}-[0-9a-f]{4}-4[0-9a-f]{3}-[89ab][0-9a-f]{3}-[0-9a-f]{12}$/, 
    'Should match UUID v4 pattern');
});

// ==================== UUID v1 Tests ====================

test('v1() generates valid UUID', () => {
  const uuid = v1();
  assertTrue(isValid(uuid), 'Generated UUID should be valid');
  assertEqual(getVersion(uuid), 1, 'Should be version 1');
});

test('v1() generates unique UUIDs', () => {
  const uuids = new Set<string>();
  for (let i = 0; i < 100; i++) {
    uuids.add(v1());
  }
  assertEqual(uuids.size, 100, 'All 100 UUIDs should be unique');
});

test('v1() generates correct format', () => {
  const uuid = v1();
  assertMatch(uuid, /^[0-9a-f]{8}-[0-9a-f]{4}-1[0-9a-f]{3}-[89ab][0-9a-f]{3}-[0-9a-f]{12}$/, 
    'Should match UUID v1 pattern');
});

// ==================== Compact UUID Tests ====================

test('compact() generates valid compact UUID', () => {
  const compactUuid = compact();
  assertEqual(compactUuid.length, 32, 'Compact UUID should be 32 characters');
  assertMatch(compactUuid, /^[0-9a-f]{32}$/, 'Should be lowercase hex');
});

test('compact() with version 1', () => {
  const compactUuid = compact(1);
  assertEqual(compactUuid.length, 32, 'Compact UUID should be 32 characters');
});

test('compact() generates unique values', () => {
  const uuids = new Set<string>();
  for (let i = 0; i < 100; i++) {
    uuids.add(compact());
  }
  assertEqual(uuids.size, 100, 'All 100 compact UUIDs should be unique');
});

// ==================== Uppercase UUID Tests ====================

test('uppercase() generates uppercase UUID', () => {
  const upper = uppercase();
  assertMatch(upper, /^[0-9A-F]{8}-[0-9A-F]{4}-[0-9A-F]{4}-[0-9A-F]{4}-[0-9A-F]{12}$/, 
    'Should be uppercase');
});

// ==================== Validation Tests ====================

test('isValid() returns true for valid UUID v4', () => {
  assertTrue(isValid('550e8400-e29b-41d4-a716-446655440000'), 'Should be valid');
});

test('isValid() returns true for valid UUID v1', () => {
  assertTrue(isValid('a0eebc99-9c0b-4ef8-bb6d-6bb9bd380a11'), 'Should be valid');
});

test('isValid() returns false for invalid UUID', () => {
  assertFalse(isValid('invalid'), 'Should be invalid');
  assertFalse(isValid(''), 'Empty string should be invalid');
  assertFalse(isValid('550e8400-e29b-41d4-a716-44665544'), 'Too short');
  assertFalse(isValid('550e8400-e29b-41d4-a716-44665544000g'), 'Invalid character');
});

test('isValid() with version check', () => {
  assertTrue(isValid('550e8400-e29b-41d4-a716-446655440000', 4), 'Should be valid v4');
  assertFalse(isValid('550e8400-e29b-41d4-a716-446655440000', 1), 'Should not be v1');
});

test('isValid() handles null and undefined', () => {
  assertFalse(isValid(null as any), 'null should be invalid');
  assertFalse(isValid(undefined as any), 'undefined should be invalid');
});

// ==================== Parsing Tests ====================

test('parse() returns correct version and variant', () => {
  const parsed = parse('550e8400-e29b-41d4-a716-446655440000');
  assertTrue(parsed !== null, 'Should parse successfully');
  assertEqual(parsed!.version, 4, 'Should be version 4');
  assertEqual(parsed!.variant, 1, 'Should be RFC 4122 variant');
});

test('parse() returns null for invalid UUID', () => {
  const parsed = parse('invalid');
  assertEqual(parsed, null, 'Should return null for invalid UUID');
});

test('parse() extracts timestamp for v1', () => {
  const uuid = v1();
  const parsed = parse(uuid);
  assertTrue(parsed !== null, 'Should parse successfully');
  assertTrue(parsed!.timestamp instanceof Date, 'Should have timestamp');
});

// ==================== Version and Variant Tests ====================

test('getVersion() returns correct version', () => {
  assertEqual(getVersion('550e8400-e29b-41d4-a716-446655440000'), 4, 'Should be version 4');
  assertEqual(getVersion('a0eebc99-9c0b-4ef8-bb6d-6bb9bd380a11'), 1, 'Should be version 1');
  assertEqual(getVersion('invalid'), -1, 'Invalid UUID returns -1');
});

test('getVariant() returns correct variant', () => {
  assertEqual(getVariant('550e8400-e29b-41d4-a716-446655440000'), 1, 'Should be RFC 4122');
  assertEqual(getVariant('invalid'), -1, 'Invalid UUID returns -1');
});

// ==================== Format Conversion Tests ====================

test('toCompact() converts standard to compact', () => {
  const compact = toCompact('550e8400-e29b-41d4-a716-446655440000');
  assertEqual(compact, '550e8400e29b41d4a716446655440000', 'Should remove hyphens');
});

test('toCompact() returns null for invalid UUID', () => {
  assertEqual(toCompact('invalid'), null, 'Should return null');
});

test('toStandard() converts compact to standard', () => {
  const standard = toStandard('550e8400e29b41d4a716446655440000');
  assertEqual(standard, '550e8400-e29b-41d4-a716-446655440000', 'Should add hyphens');
});

test('toStandard() returns null for invalid compact', () => {
  assertEqual(toStandard('invalid'), null, 'Should return null');
  assertEqual(toStandard('550e8400'), null, 'Too short');
});

test('equals() compares UUIDs case-insensitively', () => {
  assertTrue(
    equals('550e8400-e29b-41d4-a716-446655440000', '550E8400-E29B-41D4-A716-446655440000'),
    'Should be equal (case insensitive)'
  );
  assertFalse(
    equals('550e8400-e29b-41d4-a716-446655440000', '550e8400-e29b-41d4-a716-446655440001'),
    'Different UUIDs should not be equal'
  );
  assertFalse(equals('invalid', '550e8400-e29b-41d4-a716-446655440000'), 'Invalid UUIDs not equal');
});

// ==================== Nano ID Tests ====================

test('nanoId() generates correct length', () => {
  assertEqual(nanoId().length, 21, 'Default length should be 21');
  assertEqual(nanoId(10).length, 10, 'Custom length should work');
  assertEqual(nanoId(0).length, 0, 'Zero length should work');
});

test('nanoId() generates unique values', () => {
  const ids = new Set<string>();
  for (let i = 0; i < 100; i++) {
    ids.add(nanoId());
  }
  assertEqual(ids.size, 100, 'All 100 Nano IDs should be unique');
});

test('nanoId() uses URL-safe characters', () => {
  const id = nanoId(100);
  assertMatch(id, /^[A-Za-z0-9_-]+$/, 'Should only contain URL-safe characters');
});

// ==================== Short ID Tests ====================

test('shortId() generates correct length', () => {
  assertEqual(shortId().length, 8, 'Default length should be 8');
  assertEqual(shortId(16).length, 16, 'Custom length should work');
});

test('shortId() generates unique values', () => {
  const ids = new Set<string>();
  for (let i = 0; i < 100; i++) {
    ids.add(shortId());
  }
  assertEqual(ids.size, 100, 'All 100 short IDs should be unique');
});

test('shortId() uses alphanumeric characters', () => {
  const id = shortId(100);
  assertMatch(id, /^[A-Za-z0-9]+$/, 'Should only contain alphanumeric characters');
});

// ==================== Random String Tests ====================

test('randomString() generates correct length', () => {
  assertEqual(randomString(10).length, 10, 'Should generate correct length');
  assertEqual(randomString(0).length, 0, 'Zero length should work');
});

test('randomString() uses specified charset', () => {
  const str = randomString(100, CHARSETS.DIGITS);
  assertMatch(str, /^[0-9]+$/, 'Should only contain digits');
});

test('randomString() generates unique values', () => {
  const strs = new Set<string>();
  for (let i = 0; i < 100; i++) {
    strs.add(randomString(16));
  }
  assertEqual(strs.size, 100, 'All 100 strings should be unique');
});

// ==================== Random Password Tests ====================

test('randomPassword() generates correct length', () => {
  assertEqual(randomPassword().length, 16, 'Default length should be 16');
  assertEqual(randomPassword(8).length, 8, 'Minimum length should be 8');
  assertEqual(randomPassword(32).length, 32, 'Custom length should work');
});

test('randomPassword() enforces minimum length', () => {
  assertEqual(randomPassword(4).length, 8, 'Should enforce minimum length of 8');
});

test('randomPassword() contains required character types', () => {
  const password = randomPassword(16);
  assertMatch(password, /[A-Z]/, 'Should contain uppercase');
  assertMatch(password, /[a-z]/, 'Should contain lowercase');
  assertMatch(password, /[0-9]/, 'Should contain digits');
  assertMatch(password, /[!@#$%^&*()\-_=+\[\]{}|;:,.<>?]/, 'Should contain special characters');
});

test('randomPassword() generates unique passwords', () => {
  const passwords = new Set<string>();
  for (let i = 0; i < 100; i++) {
    passwords.add(randomPassword());
  }
  assertEqual(passwords.size, 100, 'All 100 passwords should be unique');
});

// ==================== Constants Tests ====================

test('CHARSETS constants are defined', () => {
  assertTrue(CHARSETS.LOWERCASE.length > 0, 'LOWERCASE should be defined');
  assertTrue(CHARSETS.UPPERCASE.length > 0, 'UPPERCASE should be defined');
  assertTrue(CHARSETS.DIGITS.length > 0, 'DIGITS should be defined');
  assertTrue(CHARSETS.SPECIAL.length > 0, 'SPECIAL should be defined');
  assertTrue(CHARSETS.HEX.length > 0, 'HEX should be defined');
  assertTrue(CHARSETS.ALPHANUMERIC.length > 0, 'ALPHANUMERIC should be defined');
  assertTrue(CHARSETS.URL_SAFE.length > 0, 'URL_SAFE should be defined');
});

// ==================== Namespace Tests ====================

test('UuidUtils namespace exports all functions', () => {
  assertTrue(typeof UuidUtils.v4 === 'function', 'v4 should be exported');
  assertTrue(typeof UuidUtils.v1 === 'function', 'v1 should be exported');
  assertTrue(typeof UuidUtils.compact === 'function', 'compact should be exported');
  assertTrue(typeof UuidUtils.uppercase === 'function', 'uppercase should be exported');
  assertTrue(typeof UuidUtils.isValid === 'function', 'isValid should be exported');
  assertTrue(typeof UuidUtils.parse === 'function', 'parse should be exported');
  assertTrue(typeof UuidUtils.getVersion === 'function', 'getVersion should be exported');
  assertTrue(typeof UuidUtils.getVariant === 'function', 'getVariant should be exported');
  assertTrue(typeof UuidUtils.toCompact === 'function', 'toCompact should be exported');
  assertTrue(typeof UuidUtils.toStandard === 'function', 'toStandard should be exported');
  assertTrue(typeof UuidUtils.equals === 'function', 'equals should be exported');
  assertTrue(typeof UuidUtils.nanoId === 'function', 'nanoId should be exported');
  assertTrue(typeof UuidUtils.shortId === 'function', 'shortId should be exported');
  assertTrue(typeof UuidUtils.randomString === 'function', 'randomString should be exported');
  assertTrue(typeof UuidUtils.randomPassword === 'function', 'randomPassword should be exported');
});

// ==================== Run Tests ====================

console.log('Running UUID Utils Test Suite...\n');

// Run all tests
// (Tests are executed as they are defined above)

console.log('\n=========================');
console.log(`Total: ${passed + failed} tests`);
console.log(`Passed: ${passed}`);
console.log(`Failed: ${failed}`);
console.log('=========================');

if (failed > 0) {
  process.exit(1);
}